src/script/lua_api/l_object.cpp

```diff
@@ -1886,25 +1886,76 @@ int ObjectRef::l_get_sky(lua_State *L)
 	if (player == nullptr)
 		return 0;
 
-	SkyboxParams skybox_params = player->getSkyParams();
+	const SkyboxParams &skybox_params = player->getSkyParams();
+
+	// handle the deprecated version
+	if (!readParam<bool>(L, 2, false)) {
+		log_deprecated(L, "Deprecated call to get_sky, please check lua_api.txt");
```

Comment:
Deprecating one way of calling the function seems a bad architectural choice to me, considering that you have to handle the case anyway in the engine for a long time to not break games and mods.

Surely you can remove the deprecated way of calling the function at one point, but then developers that want to support several versions of Minetest would have to add more complexity to their own code, which makes it unlikely that they actually do.

Reply:
In addition, once you deprecate the old version of calling the function, you would have a useless first param that needs to be true at all times … OR you deprecate calling the function with a parameter at all, which leads to further deprecation notices for something that then was correct for a time.

@Zughy what is your ideal end state and how are you planning to arrive there?

Reply:
> but then developers that want to support several versions of Minetest would have to add more complexity to their own code

I expect deprecations to be removed in major version changes, meaning you can't have compatibility anyway

> If you do not deprecate calling the function with an argument, you can keep the existing way of returning arguments forever though

The current function reflects the deprecated `set_sky()` counterpart, but I don't really mind removing the deprecation message, I can't really imagine a way to make it clean in the long run anyway, since breaking the API is prohibited

> what is your ideal end state and how are you planning to arrive there?

I guess having a `get_sky()` in 6.0 only returning a table, without the need to specify anything. But who knows when 6.0 happens, *if* happens (5.5 was planned for almost a month ago), bus factors, personal interest, warts and all. I don't know if there is a doc in the repo containing a list of things to break in the next major version, but that would definitely help as people could jot down things like this in their PRs

Reply:
> I guess having a `get_sky()` in 6.0 only returning a table, without the need to specify anything.

I hope you are aware that while *this* change is backwards compatible, *that* change will not be. Ever.